packages/grid/_modules_/grid/hooks/features/columns/gridColumnsSelector.ts

```diff
@@ -9,44 +13,47 @@ export const allGridColumnsSelector = createSelector<
   GridState,
   string[],
   GridColumnLookup,
-  GridColumns
->(allGridColumnsFieldsSelector, gridColumnLookupSelector, (allFields, lookup) => {
-  return allFields.map((field) => lookup[field]);
-});
-export const visibleGridColumnsSelector = createSelector<GridState, GridColumns, GridColumns>(
-  allGridColumnsSelector,
-  (columns: GridColumns) => columns.filter((c) => c.field != null && !c.hide),
+  GridStateColDef[]
+>(allGridColumnsFieldsSelector, gridColumnLookupSelector, (allFields, lookup) =>
+  allFields.map((field) => lookup[field]),
 );
+export const visibleGridColumnsSelector = createSelector<
+  GridState,
+  GridStateColDef[],
+  GridStateColDef[]
+>(allGridColumnsSelector, (columns) => columns.filter((c) => c.field != null && !c.hide));
 
-export const gridColumnsMetaSelector = createSelector<GridState, GridColumns, GridColumnsMeta>(
-  visibleGridColumnsSelector,
-  (visibleColumns) => {
-    const positions: number[] = [];
+export const gridColumnsMetaSelector = createSelector<
+  GridState,
+  GridStateColDef[],
+  GridColumnsMeta
+>(visibleGridColumnsSelector, (visibleColumns) => {
+  const positions: number[] = [];
 
-    const totalWidth = visibleColumns.reduce((totalW, curCol) => {
-      positions.push(totalW);
-      return totalW + curCol.width!;
-    }, 0);
+  const totalWidth = visibleColumns.reduce((totalW, curCol) => {
+    positions.push(totalW);
+    return totalW + curCol.computedWidth;
+  }, 0);
```

Comment:
Maybe use the default name we use for the reducer accumulator instead of a strangely named variable?

```suggestion
  const totalWidth = visibleColumns.reduce((acc, curCol) => {
    positions.push(acc);
    return acc + curCol.computedWidth;
  }, 0);
```